tests/process_junit.py

```diff
@@ -0,0 +1,365 @@
+from __future__ import annotations
+
+import dataclasses
+import json
+import random
+import re
+from collections import defaultdict
+from dataclasses import dataclass, field
+from pathlib import Path
+from statistics import StatisticsError, mean, stdev
+from typing import Any, Dict, List, Optional, TextIO, Tuple, Type, final
+
+import click
+import lxml.etree
+
+from tests.util.misc import BenchmarkData, DataTypeProtocol, TestId
+from tests.util.time_out_assert import TimeOutAssertData
+
+supported_data_types: List[Type[DataTypeProtocol]] = [TimeOutAssertData, BenchmarkData]
+supported_data_types_by_tag: Dict[str, Type[DataTypeProtocol]] = {cls.tag: cls for cls in supported_data_types}
+
+
+@final
+@dataclass(frozen=True, order=True)
+class Result:
+    file_path: Path
+    test_path: Tuple[str, ...]
+    ids: Tuple[str, ...]
+    label: str
+    line: int = field(compare=False)
+    durations: Tuple[float, ...] = field(compare=False)
+    limit: float = field(compare=False)
+
+    def marshal(self) -> Dict[str, Any]:
+        return {
+            "file_path": self.file_path.as_posix(),
+            "test_path": self.test_path,
+            "label": self.label,
+            "duration": {
+                "all": self.durations,
+                "min": min(self.durations),
+                "max": max(self.durations),
+                "mean": mean(self.durations),
+            },
+        }
+
+    def link(self, prefix: str, line_separator: str) -> str:
+        return f"{prefix}{self.file_path.as_posix()}{line_separator}{self.line}"
+
+
+def sub(matchobj: re.Match[str]) -> str:
```

Comment:
I don't see this function used anywhere

Reply:
Yep, indeed.  Seems I made this monstrosity pointless.  Thanks.

Reply:
Removed in https://github.com/Chia-Network/chia-blockchain/pull/16831/commits/de39fc06bfd154308455936d22c4c5a7e5ce4d19.